tools/rpm2img

```diff
@@ -408,18 +408,30 @@ for side in A B ; do
   fi
   dev="BOTTLEROCKET_DATA_${side}"
   dev="${!dev}"
-  mkfs.ext4 -d "${DATA_MOUNT}" "${dev}" "${partsize["DATA-${side}"]}M"
-  echo "${UNLABELED}" | debugfs -w -f - "${dev}"
+
+  # Only create the filesystem if the size is big enough to allocate enough
+  # inode-per-blocks for when we resize the fs to fit a bigger data partition.
+  # If we create the data fs now with the smaller size, the inode-per-block
+  # count will be capped and when we resize the filesystem to a larger size
+  # later, the total inode count will be capped based on the smaller initial
+  # filesystem creation size.
+  # If we don't create the filesystem now, we'll create the filesystem at
+  # runtime after the data partition has been resized.
+  if [ "${partsize["DATA-${side}"]}" -ge 1024 ]; then
+    kfs.ext4 -b 4096 -O resize_inode -E "resize=268435456" -i 16384 -I 256  -m 0 -d "${DATA_MOUNT}" "${dev}" "${partsize["DATA-${side}"]}M"
+      echo "${UNLABELED}" | debugfs -w -f - "${dev}"
+  fi
 done
 
 case "${PARTITION_PLAN}" in
   split)
-    # Only create data partition B on data image in "split" image layout configurations
+    # Only create DATA-B on data image in "split" image layout configurations
     dd if="${BOTTLEROCKET_DATA_B}" of="${DATA_IMAGE}" conv=notrunc bs=1M seek="${partoff[DATA-B]}"
 
     dd if="${BOTTLEROCKET_DATA_A}" of="${OS_IMAGE}" conv=notrunc bs=1M seek="${partoff[DATA-A]}"
```

Comment:
We can omit this since it should be a zero-length file in the split case.

But then we're sort of back to needing only one BOTTLEROCKET-DATA filesystem images in all cases.